storage-access.bs

```diff
@@ -182,14 +176,7 @@ When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>ha
 1. If |doc|'s [=Document/origin=] is [=same origin=] with the [=top-level origin=] of |doc|'s [=relevant settings object=], [=/resolve=] |p| with true and return |p|.
 1. Let |key| be the result of [=generate a partitioned storage key|generating a partitioned storage key=] from |doc|.
 1. If |key| is failure, [=resolve=] |p| with false and return |p|.
-1. Run these steps [=in parallel=]:
-    1. Let |map| be the result of [=obtain the storage access map|obtaining the storage access map=] for |doc|.
-    1. Let |flag set| be the result of [=obtain a storage access flag set|obtaining the storage access flag set=] with |key| from |map|.
-    1. If |flag set|'s [=was expressly denied storage access flag=] is set, [=queue a global task=] on the [=permission task source=] given |global| to [=/resolve=] |p| with false, and abort these steps.
-    1. If |flag set|'s [=has storage access flag=] is set, [=queue a global task=] on the [=permission task source=] given |global| to [=/resolve=] |p| with true, and abort these steps.
-    1. Let |hasAccess| be [=a new promise=].
-    1. [=Determine the storage access policy=] with |key|, |doc| and |hasAccess|.
-    1. [=Queue a global task=] on the [=permission task source=] given |global| to [=/resolve=] |p| with the result of |hasAccess|.
+1. [=Resolve=] |p| with the result of running [=determine if a site has storage access=] with |key| and |doc|.
```

Comment:
So this is an observable change in that we no longer queue a task and as such the promise will resolve in the current task. E.g., if you do `enqueueMicrotask()` after this call it'll happen immediately afterwards. Does that match implementations?

Reply:
Right, looking at this again it seems like `hasStorageAccess` can now be synchronous, as it only depends on a flag stored on the environment?

This seems like a side-effect of the per-frame change more than the permissions one, so if you prefer we can queue a task here (does that need to happen in parallel?) and decide on `hasStorageAccess` later?

Reply:
No, you could queue a task from the main thread as well to preserve the timing aspects.

Reply:
Filed #146, will restore the task queuing here.